lms/templates/bundle.html

```diff
@@ -0,0 +1,47 @@
+<%def name="online_help_token()"><% return "course" %></%def>
+<%namespace name='static' file='static_content.html'/>
+<%!
+from django.utils.translation import ugettext as _
+from django.core.urlresolvers import reverse
+from six import text_type
+from urlparse import urljoin
+%>
+<%page args="bundle" expression_filter="h"/>
+<article class="bundle" id="${bundle.get('uuid')}" role="region" aria-label="${bundle.get('title')}">
+    <%
+        images = []
+        orgs = []
+
+        for journal in bundle['journals']:
+            images.append(journal.get('card_image_url', ''))
+            orgs.append(journal.get('organization', ''))
+
+        for course in bundle['courses']:
+            images.append(course.get('img', {}).get('src', ''))
+            orgs.append(journal.get('partner', ''))
+
+        card_img = next((img for img in images if img), '')
+        organization = next((org for org in orgs if org), '')
+    %>
+    <a class="journals-listing-item" href="${reverse('openedx.journals.bundle_about', kwargs={'bundle_uuid':bundle['uuid']})}">
+        <header class="journal-image">
+            <div class="cover-image">
+            <img src="${card_img}" alt="${bundle.get('title')}" />
```

Comment:
Not sure if this is working as intended. I have a bundle with a journal with no card image but a course that has a card image. I expected to see the courses card image but I see no image.

Reply:
possibly related, seeing this error when I click on the bundle card:
```
edx.devstack.lms | Internal Server Error: /bundles/edf824b9-ecd6-4983-a16e-a3034b01d98a/about
edx.devstack.lms | Traceback (most recent call last):
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
edx.devstack.lms |     response = get_response(request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
edx.devstack.lms |     response = self._get_response(request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
edx.devstack.lms |     response = self.process_exception_by_middleware(e, request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
edx.devstack.lms |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 185, in inner
edx.devstack.lms |     return func(*args, **kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/openedx/features/journals/views/marketing.py", line 23, in bundle_about
edx.devstack.lms |     return render_to_response('journals/bundle_about.html', context)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 169, in render_to_response
edx.devstack.lms |     return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 159, in render_to_string
edx.devstack.lms |     return template.render(dictionary, request)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 59, in render
edx.devstack.lms |     return self.mako_template.render_unicode(**context_dictionary)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/template.py", line 454, in render_unicode
edx.devstack.lms |     as_unicode=True)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
edx.devstack.lms |     **_kwargs_for_callable(callable_, data))
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
edx.devstack.lms |     _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
edx.devstack.lms |     callable_(context, *args, **kwargs)
edx.devstack.lms |   File "/tmp/mako_lms/59c23f148a89e6cd85b4028af7bd18c2/main.html.py", line 352, in render_body
edx.devstack.lms |     __M_writer(filters.decode.utf8(self.body()))
edx.devstack.lms |   File "/tmp/mako_lms/59c23f148a89e6cd85b4028af7bd18c2/journals/bundle_about.html.py", line 91, in render_body
edx.devstack.lms |     course_img = course.get('image', {}).get('src', '')
edx.devstack.lms | AttributeError: 'NoneType' object has no attribute 'get'
edx.devstack.lms | 2018-05-03 16:01:21,677 ERROR 130 [django.request] exception.py:135 - Internal Server Error: /bundles/edf824b9-ecd6-4983-a16e-a3034b01d98a/about
edx.devstack.lms | Traceback (most recent call last):
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
edx.devstack.lms |     response = get_response(request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
edx.devstack.lms |     response = self._get_response(request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
edx.devstack.lms |     response = self.process_exception_by_middleware(e, request)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
edx.devstack.lms |     response = wrapped_callback(request, *callback_args, **callback_kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 185, in inner
edx.devstack.lms |     return func(*args, **kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/openedx/features/journals/views/marketing.py", line 23, in bundle_about
edx.devstack.lms |     return render_to_response('journals/bundle_about.html', context)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 169, in render_to_response
edx.devstack.lms |     return HttpResponse(render_to_string(template_name, dictionary, namespace, request), **kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/shortcuts.py", line 159, in render_to_string
edx.devstack.lms |     return template.render(dictionary, request)
edx.devstack.lms |   File "/edx/app/edxapp/edx-platform/common/djangoapps/edxmako/template.py", line 59, in render
edx.devstack.lms |     return self.mako_template.render_unicode(**context_dictionary)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/template.py", line 454, in render_unicode
edx.devstack.lms |     as_unicode=True)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 829, in _render
edx.devstack.lms |     **_kwargs_for_callable(callable_, data))
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 864, in _render_context
edx.devstack.lms |     _exec_template(inherit, lclcontext, args=args, kwargs=kwargs)
edx.devstack.lms |   File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/mako/runtime.py", line 890, in _exec_template
edx.devstack.lms |     callable_(context, *args, **kwargs)
edx.devstack.lms |   File "/tmp/mako_lms/59c23f148a89e6cd85b4028af7bd18c2/main.html.py", line 352, in render_body
edx.devstack.lms |     __M_writer(filters.decode.utf8(self.body()))
edx.devstack.lms |   File "/tmp/mako_lms/59c23f148a89e6cd85b4028af7bd18c2/journals/bundle_about.html.py", line 91, in render_body
edx.devstack.lms |     course_img = course.get('image', {}).get('src', '')
edx.devstack.lms | AttributeError: 'NoneType' object has no attribute 'get'
edx.devstack.lms | [03/May/2018 16:01:21] "GET /bundles/edf824b9-ecd6-4983-a16e-a3034b01d98a/about HTTP/1.1" 500 414340
```

Reply:
Found the issue.` course.get('image')` was returning None so the second `get` failed. Fixed in new commit.